protobuf/src/main/java/io/grpc/protobuf/ProtoUtils.java

```diff
@@ -107,7 +107,8 @@ private T parseFrom(InputStream stream) throws InvalidProtocolBufferException {
    */
   public static <T extends Message> Metadata.Key<T> keyForProto(final T instance) {
     return Metadata.Key.of(
-        instance.getDescriptorForType().getFullName() + Metadata.BINARY_HEADER_SUFFIX,
+        instance.getDescriptorForType().getFullName().replace('.', '-')
```

Comment:
Was this change verified at all? Any existing users of keyForProto will not be able to interoperate. We need to make sure the behavior you have here is the cross-language behavior.